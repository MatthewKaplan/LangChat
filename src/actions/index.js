export const changeEmail = text => ({
	type: 'CHANGE_EMAIL',
	text
});

export const changePassword = text => ({
	type: 'CHANGE_PASSWORD',
	text
});

export const addMessage = message => ({
	type: 'ADD_MESSAGE',
	payload: { message }
});
