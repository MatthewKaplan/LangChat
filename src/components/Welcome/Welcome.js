import React, { Component } from 'react';
import { Text, View, StyleSheet } from 'react-native';
import NavDrawer from '../NavDrawer/NavDrawer';

export default class Welcome extends Component {
	render () {
		return (
			<View style={styles.container}>
				<NavDrawer>
					<Text>Welcome back {this.props.username}!</Text>
				</NavDrawer>
			</View>
		);
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1.0,
		position: 'relative'
	},
	text: {
		fontSize: 50
	},
	drawerContainer: {
		flex: 1.0,
		zIndex: 5
	},
	welcomeContainer: {
		position: 'absolute',
		flex: 1.0,
		zIndex: 1
	}
});
