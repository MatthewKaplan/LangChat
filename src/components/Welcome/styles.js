import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
	container: {
    flex: 1.0,

  
	},
	welcomeText: {
		fontSize: 18,
		justifyContent: 'center',
    marginTop: 50,
    fontSize: 24,
    textAlign: 'center'
    
	},
	offColor: {
		color: '#3B5998',
		fontWeight: 'bold',
		fontSize: 26
	},
	imageStyle: {
		height: '40%',
		width: '100%'
	}
});
