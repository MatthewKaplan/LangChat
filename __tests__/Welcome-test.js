// __tests__/Welcome-test.js
import React from 'react';
import { Welcome, mapStateToProps, mapDispatchToProps } from '../src/components/Welcome/Welcome';
import * as actions from '../src/actions/index';
import { shallow } from 'enzyme';

const mockUser = {
	country_of_origin: 'United States',
	email: 'Mkaplan@mail.com',
	first_name: 'Matthew',
	id: 10,
	is_active: true,
	last_name: 'Kaplan',
	password: 'pbkdf2_sha256$150000$AdrUegMOrPIc$MFeO3KKmLLesav6R9k4Zu5kmSXh70LdNY8RMf26CzH8=',
	token: 'e85595630776825bdf128ae87da88432bc02e9c3',
	username: 'Mkaplan'
};

describe('Welcome', () => {
	let wrapper, instance;

	beforeEach(() => {
		wrapper = shallow(
			<Welcome user={mockUser} />
		);
		instance = wrapper.instance();
	});

	it(`matches snapshot`, () => {
		expect(wrapper).toMatchSnapshot();
  });
});

describe('mapStateToProps', () => {
	it('should return an object', () => {
		const mockData = {
			user: ''
		};
		const expected = {
			user: ''
		};

		const mockProps = mapStateToProps(mockData);
		expect(mockProps).toEqual(expected);
	});
});