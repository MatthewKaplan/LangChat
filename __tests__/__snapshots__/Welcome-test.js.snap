// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome matches snapshot 1`] = `ShallowWrapper {}`;
